src/devicewrapper.cpp

```diff
@@ -0,0 +1,66 @@
+/* ****************************************************************************
+*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+*
+*
+* This file is part of Microsoft R Host.
+*
+* Microsoft R Host is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Microsoft R Host is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Microsoft R Host.  If not, see <http://www.gnu.org/licenses/>.
+*
+* ***************************************************************************/
+
+#include "devicewrapper.h"
+#include "msvcrt.h"
+
+namespace rhost {
+    namespace grdevices {
+        devicewrapper::devicewrapper() {
+            _ver = R_GE_getVersion();
+        }
+
+
+        devicewrapper::devicewrapper(pDevDesc dd) {
+            _ver = R_GE_getVersion();
+            switch (_ver) {
```

Comment:
Maybe union is best after all, if we end up with a field for v10. This will only grow unnecessarily if it's not a union.

---
In reply to: [113539632](https://github.com/Microsoft/R-Host/pull/133#discussion_r113539632) [](ancestors = 113539632)